pkg/identity/numericidentity.go

```diff
@@ -77,35 +77,49 @@ const (
 	// ReservedIdentityRemoteNode is the identity given to all nodes in
 	// local and remote clusters except for the local node.
 	ReservedIdentityRemoteNode
+)
 
-	// --------------------------------------------------------------
-	// Special identities for well-known cluster components
-
+// Special identities for well-known cluster components
+// Each component has two identities. The first one is used for Kubernetes <1.21
+// or when the NamespaceDefaultLabelName feature gate is disabled. The second
+// one is used for Kubernets >= 1.21 and when the NamespaceDefaultLabelName is
+// enabled.
+const (
 	// ReservedETCDOperator is the reserved identity used for the etcd-operator
 	// managed by Cilium.
-	ReservedETCDOperator NumericIdentity = 100
+	ReservedETCDOperator NumericIdentity = iota + 100
 
 	// ReservedCiliumKVStore is the reserved identity used for the kvstore
 	// managed by Cilium (etcd-operator).
-	ReservedCiliumKVStore NumericIdentity = 101
+	ReservedCiliumKVStore
 
 	// ReservedKubeDNS is the reserved identity used for kube-dns.
-	ReservedKubeDNS NumericIdentity = 102
+	ReservedKubeDNS
 
 	// ReservedEKSKubeDNS is the reserved identity used for kube-dns on EKS
-	ReservedEKSKubeDNS NumericIdentity = 103
+	ReservedEKSKubeDNS
 
 	// ReservedCoreDNS is the reserved identity used for CoreDNS
-	ReservedCoreDNS NumericIdentity = 104
+	ReservedCoreDNS
 
 	// ReservedCiliumOperator is the reserved identity used for the Cilium operator
-	ReservedCiliumOperator NumericIdentity = 105
+	ReservedCiliumOperator
 
 	// ReservedEKSCoreDNS is the reserved identity used for CoreDNS on EKS
-	ReservedEKSCoreDNS NumericIdentity = 106
+	ReservedEKSCoreDNS
 
 	// ReservedCiliumEtcdOperator is the reserved identity used for the Cilium etcd operator
-	ReservedCiliumEtcdOperator NumericIdentity = 107
+	ReservedCiliumEtcdOperator
+
+	// Second identities for all above components
```

Comment:
Somehow I feel there should be a gap in numbering between the old and new identities for some future expansion, maybe even start the 2nd versions from `200` to identify them as a group in logs etc. Could @aanm weigh in on this topic?

Reply:
@jrajahalme although they look like a 2nd versions, I see them as just a new set of labels that are accounted as part of well known identities thus I don't think it's worth having them start at `200`  (we only have 256 something reserved identities)

Reply:
@aanm But they will occupy a numerical value from the well known identities space, no?

Reply:
So if we want to add a new reserved identity, then it will go after the second identities set. I guess it's fine?

Reply:
Existing well-known identities before this patch will have a gap between the first and second one (107 & 115 for CiliumEtcdOperator), new identities added after this PR will have both of them consecutive (116 & 117 for instance).